python/tk_houdini/utils/parm_template_wrappers.py

```diff
@@ -93,25 +163,60 @@ def index_of_template(self, template_name):
         """
         return self.__child_names.index(template_name)
 
+    def extend(self, parms):
+        """
+        Add multiple templates to the child templates list..
+
+        :param list(Parm)) parms: The list of Parm
+            instances to add.
+        """
+        for parm in parms:
+            self.append(parm)
+
     def extend_templates(self, templates):
         """
         Add multiple templates to the child templates list..
 
         :param list(hou.ParmTemplate) templates: The list of template
             instances to add.
         """
-        template_names = (template.name() for template in templates)
-        child_templates = (Parm.build_parm(template) for template in templates)
-        self.__child_names.extend(template_names)
-        self.__children.extend(child_templates)
+        child_templates = (
+            Parm.build_parm(template, parent=self) for template in templates
+        )
+        self.extend(templates)
```

Comment:
Shouldn't this be:
```python
        self.extend(child_templates)
```

You could also just shorten it to:
```suggestion
        self.extend(
            Parm.build_parm(template, parent=self) for template in templates
        )
```